Words Game/ps3.py

```
@@ -0,0 +1,586 @@
+        else:
+            return False
+
+    # If there is not the wildcard(asterisk) in the word, checks if the word exists.
+    elif word.lower() not in word_list:
+        return False  # If the word does not exist, returns False.
+
+    # Converts the word to dictionary(string -> int), same as hand.
+    word_frequency_dict = get_frequency_dict(word.lower())
+
+    # If the number of occurrences of a letter more than the number in the hand, returns False.
+    for key in word_frequency_dict:
+        if word_frequency_dict[key] > hand.get(key, 0):
+            return False
+
+    # Otherwise, returns True.
+    return True
+
+
+def calculate_handlen(hand):
+    """ 
+    Returns the length (number of letters) in the current hand.
+    
+    hand: dictionary (string-> int)
+    returns: integer
+    """
+
+    length = 0
+    # Counts the number of occurrences of each letter in the word and adds together.
+    for letter in hand.keys():
+        length += hand.get(letter, 0)
+
+    return length
+
+
+def ask_user(prompt, **kwargs):
+    """
+    Asks user for entering the corresponding values and checks the inputs depending on the parameters.
+    Returns the correct input user entered.
+
+    prompt: string, helpful input message.
+    type: string, the input format(int, float, single(only one character)), default - str.
+    bool: bool, indicates if the input should be only 'Yes' or 'No', default - False.
+    returns: string.
+    """
+
+    # Allows to enter value until it gets the right one.
+    while True:
+        answer = input(prompt)
+        try:
+            for key, value in kwargs.items():
+                # If the function calls with validator or values parameter, checks the input.
+                if (key == 'validator' and not value(answer) or
+                        (key == 'values' and answer.lower() not in value)):
+                    # If an error detected, prints warning message.
+                    raise ValueError
```

You should raise errors only in case of something unexpected. Here it's absolutely expected that user can enter invalid input. Please - remove using try-except from this function